pkgs/by-name/cr/crabfit-api/package.nix

```
@@ -0,0 +1,63 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  protobuf,
+  openssl,
+  sqlite,
+  stdenv,
+  darwin,
+  adaptor ? "sql",
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "crabfit-api";
+  version = "0-unstable-2023-08-02";
+
+  src = fetchFromGitHub {
+    owner = "GRA0007";
+    repo = "crab.fit";
+    rev = "628f9eefc300bf1ed3d6cc3323332c2ed9b8a350";
+    hash = "sha256-jy8BrJSHukRenPbZHw4nPx3cSi7E2GSg//WOXDh90mY=";
+  };
+
+  sourceRoot = "source/api";
+
+  patches = [ ./01-listening-address.patch ];
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "google-cloud-0.2.1" = "sha256-3/sUeAXnpxO6kzx5+R7ukvMCEM001VoEPP6HmaRihHE=";
+    };
+  };
```

I'm not too familiar with cargo/rust, but why does using the upstream lockfile not work here?
```suggestion
  cargoLock = {
    lockFile = ${src}/Cargo.lock;
    outputHashes = {
      "google-cloud-0.2.1" = "sha256-3/sUeAXnpxO6kzx5+R7ukvMCEM001VoEPP6HmaRihHE=";
    };
  };
```
since upstream has one https://github.com/GRA0007/crab.fit/blob/main/api/Cargo.lock